doc/rfd/rfd/0009/README.md

```diff
@@ -0,0 +1,176 @@
+distribution from them without creating an all-or-nothing "super-chart" is
+difficult. One needs to make sure that deployment and automation scripts are
+interoperable between charts, repos, repos' histories, and that they can be
+reused and extended downstream as needed.
+
+At the same time one needs to create, reuse, and extend test environments across
+charts, and between upstream and downstream.
+
+The following spec is an interface specification for sharing deployment and
+automation scripts of Helm charts and utility applications.
+
+
+## Specification
+
+### Version `0.1`.
+
+
+The interface consists in the following structure of folders and executables,
+together with their expected use:
+
+```
+├── clean/*
+├── chart/*
+├── configure/*
+│   └── patches/*
+├── install/*
+├── wait/*
+├── include/*
+│
+├── one-offs/*
+│
+├── include/*
+├── example-values/*
+│
+└── spec_ver
+```
+
+This structure as a whole is encouraged to be published together with the Helm
+chart. This can be for example as artifacts of a chart release, or present in
+the chart repository, but we recommend that this files are bundled inside of the
+chart: this simplifies consumption downstream.
+
+While targets are allowed to take arguments and options on execution besides the
+expected default behaviour defined here, this is discouraged.
+
+Each target is expected to be a folder (E.g: `install/`) containing executable
+files starting with digits (E.g: `10_do-foo.sh`, `20_do-bar.rb`).
+
+Upon target call, all the executable files starting with digits will get
+executed in alphanumeric order.
+It is allowed for the folder to contain other files/folders besides those
+starting with digits. Those would not be automatically executed upon target
+call (e.g: readme files, local include folders, etc).
+  
+Deployment and automation scripts will expose the following targets:
+
+- `chart`: Obtains the latest public release of the chart at hand, and puts it
+  already decompressed in `$pwd/$our-chart/chart/`. `$our-chart` being the name
+  of the chart at hand (e.g: kubecf).
+
+- `configure`: Outcome is a valid `$pwd/$our-chart/values.yaml` file for the
+  chart at hand. This target accepts a list of alphanumerically ordered subsets
+  of `values.yaml`, inside the folder `configure/patches/`, which get merged to
+  output the final values.yaml. This allows for reuse of existing example
+  values, and for downstream tools to inject their own yaml subsets as needed.
+
+- `install`: Takes a `$pwd/$our-chart/values.yaml` and the chart contained in
+  `$pwd/$our-chart/chart/` and installs the Helm Chart at hand. 
+
+- `wait`: Will wait until deployment of the chart is up and running.
+
+- `upgrade`: Takes a `$pwd/values.yaml` and a chart contained in
+  `$pwd/$our-chart/chart/` and upgrades the Helm Chart at hand. The upgrade must
+  work for both same version and different versions.
+
+- `clean`: Removes all possible objects created by any target from the cluster.
```

Comment:
Can you split this into `clean` and `delete`?
`clean` can remove objects from the cluster, and `delete` can delete the chart.
I usually want to remove things from the cluster quite often, but rarely want to delete the chart from my local disk.

Reply:
just when you were reviewing, I tackled Mario's review and reworded `clean` and added a `clobber` target that leaves the cluster pristine. Would that be ok?

Reply:
renamed `clean` to `uninstall` in https://github.com/cloudfoundry-incubator/kubecf/pull/1495/commits/76c684f28d9a79c6483c15fe125b7380658fd6e9, and we have clobber nevertheless.